enumerables.rb

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/ruby
+# frozen_string_literal: true
+
+# rubocop:disable ModuleLength
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0...size).each do |i|
+        yield(self[i])
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0...size).each do |i|
+        yield(self[i], i)
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+      if instance_of?(Array)
+        temp_array = []
+        my_each do |item|
+          temp_array << item if yield(item)
+        end
+        temp_array
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |x| return false unless yield x }
+    else
+      return my_all? { |obj| obj } unless pattern
+
+      if pattern.class == Regexp
+        my_each { |x| return false unless pattern.match?(x) }
+      elsif pattern.class == Class
+        my_each { |x| return false unless x.is_a? pattern }
+      else
+        my_each { |x| return false unless x == pattern }
+      end
+    end
+    true
+  end
+
+  def my_any?
+    if block_given?
+      my_each { |x| return true unless yield x }
+    else
+      return my_all? { |obj| obj } unless pattern
+
+      if pattern.class == Regexp
+        my_each { |x| return true unless pattern.match?(x) }
+      elsif pattern.class == Class
+        my_each { |x| return true unless x.is_a? pattern }
+      else
+        my_each { |x| return true unless x == pattern }
+      end
+    end
+    false
+  end
+
+  def my_none?
+    if block_given?
+      !my_any? { |x| yield x }
+    elsif pattern
+      !my_any?(pattern)
+    else
+      my_each { |x| return false if x }
+      true
+    end
+  end
```

Comment:
* [ ] my_none? should be able to receive classes, literals, regex exp, etc as **arguments** :no_entry_sign: 

* [ ] my_none? returns true if the block never returns true for all elements
  ```ruby
    array = [1,2,5,4,7]
    true_block = proc {|num| num > 10 }
    p array.my_none?(&true_block) == array.none?(&true_block) # false
  ```